editor/src/messages/portfolio/document/document_message_handler.rs

```diff
@@ -297,7 +297,8 @@ impl MessageHandler<DocumentMessage, DocumentMessageData<'_>> for DocumentMessag
 					.collect();
 
 					let id = NodeId(generate_uuid());
-					let insert_index = -1;
+					let selected_layer_index = parent.children(self.metadata()).collect::<Vec<_>>().iter().position(|&sibling| sibling == layer).unwrap_or(0);
+					let insert_index = if (selected_layer_index as i64 - 1) < 0 { -1 } else { selected_layer_index as isize };
```

Comment:
Since `-1` and `0` play the same role here, you may simply use `let insert_index = <...usize> as isize`.

Reply:
you can simply use `let insert_index = parent.children(self.metadata()).collect::<Vec<_>>().iter().position(|&sibling| sibling == layer).unwrap_or(0) as isize;`